LibGit2Sharp/ObjectDatabase.cs

```
@@ -645,45 +645,53 @@ public virtual MergeTreeResult MergeCommits(Commit ours, Commit theirs, MergeTre
+        /// Packs objects in the <see cref="ObjectDatabase"/>, and write a pack (.pack) and index (.idx) 
+        /// files for them to the <paramref name="packDirectory"/>
         /// </summary>
+        /// <param name="packDirectory">Output directory for pack and idx files.</param>
         /// <param name="options">Packing options</param>
-        /// <param name="packDelegate">Packing action</param>
         /// <returns>Packing results</returns>
-        public virtual PackBuilderResults Pack(PackBuilderOptions options, Action<PackBuilder> packDelegate)
+        public virtual PackBuilderResults Pack(string packDirectory, PackBuilderOptions options)
         {
-            return InternalPack(options, packDelegate);
+            Ensure.ArgumentNotNullOrEmptyString(packDirectory, "packDirectory");
+            if (!Directory.Exists(packDirectory))
+            {
+                throw new DirectoryNotFoundException("The Directory " + packDirectory + " does not exist.");
+            }
+
+            return InternalPack(options, builder => builder.Write(packDirectory));
         }
 
         /// <summary>
-        /// Packs objects in the <see cref="ObjectDatabase"/> and write a pack (.pack) and index (.idx) files for them.
+        /// Packs objects in the <see cref="ObjectDatabase"/> and use the <paramref name="writeAction"/>
+        /// to write the results.
         /// For internal use only.
         /// </summary>
         /// <param name="options">Packing options</param>
-        /// <param name="packDelegate">Packing action</param>
+        /// <param name="writeAction">Writing action action</param>
         /// <returns>Packing results</returns>
-        private PackBuilderResults InternalPack(PackBuilderOptions options, Action<PackBuilder> packDelegate)
+        private PackBuilderResults InternalPack(PackBuilderOptions options, Action<PackBuilder> writeAction)
         {
             Ensure.ArgumentNotNull(options, "options");
```

Not a change introduced here, but since we're churning the API anyway....why are we throwing here instead of using a default `PackBuilderOptions`?